dali/kernels/slice/slice_cpu.h

```diff
@@ -28,31 +29,242 @@ namespace kernels {
 
 namespace detail {
 
-template <typename OutputType, typename InputType, int Dims>
+/**
+ * @brief Fills output with nchannel values repeatedly
+ */
+template <typename T>
+void Fill(T *output, const T *fill_values, int64_t npixels, int64_t nchannels) {
+  int64_t n = npixels * nchannels;
+  int64_t i = 0;
+  for (; i < nchannels; i++)
+    output[i] = fill_values[i];
+  for (; i < n; i++)
+    output[i] = output[i - nchannels];
+}
+
+inline std::tuple<int64_t, int64_t, int64_t> CalcPadCopyExtents(int64_t anchor,
+                                                                int64_t in_extent,
+                                                                int64_t out_extent) {
+  int64_t pad_before = std::max<int64_t>(0, std::min(out_extent, -anchor));
```

Comment:
Do we assume that anchor can be negative?

Reply:
Maybe call `abs` instead?

Reply:
Yes, the anchor can be negative.
This actually takes care of both out of bounds cases:
- left side: `pad_before` will be `-anchor`
- right side: `pad_before` will be `out_extent` (the whole slice is padding since it's completely out of bounds)

Reply:
Anchor can be negative, and no - we can't use abs here.

Reply:
I'll rewrite as ```int64_t pad_before = std::min(out_extent, std::max<int64_t>(0, -anchor));``` as @mzient suggested